src/api/models/iuf/manifestDataValidation/manifest_data_validate.go

```diff
@@ -0,0 +1,303 @@
+		}
+
+		if nexusContent[REPO_TYPE] == "hosted" {
+
+			vs.hostedRepoNames = append(vs.hostedRepoNames, nexusContent["name"].(string))
+			temp_repo_names = append(temp_repo_names, nexusContent["name"].(string))
+
+		} else if nexusContent[REPO_TYPE] == "group" {
+			group_map := nexusContent["group"].(map[string]interface{})
+
+			for _, v := range group_map {
+				memNames_array := v.([]interface{})
+				for _, m := range memNames_array {
+
+					memberRepo := m.(string)
+
+					isHostedRepo, index := mutils.StringFoundInArray(temp_repo_names, memberRepo)
+					if isHostedRepo {
+						temp_repo_names, err = mutils.Delete(temp_repo_names, index)
+
+						if err != nil {
+							fmt.Println("Repo defined in host repo is not listed in group repo")
+						}
+
+					} else {
+						return fmt.Errorf("Repo referenced in group does not match hosted repo or Hosted Repos are not listed before group repos")
+					}
+				}
+			}
+		}
+
+	}
+
+	if len(temp_repo_names) > 0 {
+		return fmt.Errorf("Repo defined in host repo is not listed in group repo")
+	}
+
+	return nil
+}
+
+// Method to process nexus blob content, returns error in case of issues
+func (vs *validators) processNexusBlob() error {
+	nb, nb_present := vs.content[NEXUS_BLOB_KEY] // extracting nexus blob key
+
+	if !nb_present {
+		return nil // nexus blob key missing
+	}
+
+	nb_map := nb.(map[string]interface{}) // assuming map, is it validated in schema??
+	file_path := nb_map[NEXUS_BLOB_PATH_KEY].(string)
+
+	exist := mutils.IsPathExist(file_path) // do we need error details??
+	if !exist {                            // if path is invalid
+		return fmt.Errorf("error in processing nexus repo file %v", file_path)
+	}
+	return nil
+}
+
+// Method to process vcs content, return error in case of issues
+func (vs *validators) processVcs() error {
+
+	vcs, vcs_present := vs.content[VCS_KEY] // extracting nexus repo key
+
+	if !vcs_present {
+		return nil // vcs key missing
+	}
+	vcs_map := vcs.(map[string]interface{})
+
+	dir_path := vcs_map[VCS_PATH_KEY].(string)
+	empty := mutils.IsEmptyDirectory(dir_path) // do we need error details??
+	if empty {                                 // if path is invalid
+		return fmt.Errorf("error in processing vcs directory %v", dir_path)
+	}
+	return nil
+}
+
+// Method to process rpm content, return error in case of issues
+func (vs *validators) processRpm() error {
+
+	rpm, rpm_present := vs.content[RPM_KEY] // extracting rpm key
+
+	if !rpm_present {
+		return nil // rpm key missing
+	}
+
+	rpm_array := rpm.([]interface{})
+
+	for _, rpm := range rpm_array {
+		rpm_map := rpm.(map[string]interface{})
+
+		dir_path := rpm_map[RPM_PATH_KEY].(string)
+		empty := mutils.IsEmptyDirectory(dir_path) // do we need error details??
+		if empty {                                 // if path is invalid
+			return fmt.Errorf("error in processing rpm directory %v", dir_path)
+		}
+		repoName := rpm_map[REPO_NAME].(string)
```

Comment:
Lets move it out to create a separate function func (vs *validators) validateAllRpmsHosted() error{